lib/ja_serializer/builder/relationship.ex

```diff
@@ -9,10 +9,14 @@ defmodule JaSerializer.Builder.Relationship do
   def build(%{serializer: serializer, data: data, conn: conn, opts: opts} = context) do
     case opts[:relationships] do
       false -> []
-      _ -> Enum.map serializer.relationships(data, conn), &(build(&1, context))
+      _ -> Enum.map(serializer.relationships(data, conn), &(build(&1, context)))
+           |> Enum.filter(fn r -> not empty?(r) end)
```

Comment:
to match the above syntax this can probably be rewritten as:

```elixir
|> Enum.filter(&(not empty?(&1)))
```

Reply:
@bcardarella what you suggested produces a compilation error: 
```
== Compilation error in file lib/ja_serializer/builder/relationship.ex ==
** (CompileError) lib/ja_serializer/builder/relationship.ex:13: nested captures via & are not allowed: &(not empty?(&1))
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (elixir) src/elixir_fn.erl:133: :elixir_fn.escape/3
    (stdlib) lists.erl:1354: :lists.mapfoldl/3
    (stdlib) lists.erl:1355: :lists.mapfoldl/3
    (elixir) src/elixir_fn.erl:133: :elixir_fn.escape/3
    (elixir) src/elixir_fn.erl:105: :elixir_fn.capture_expr/5
```
I will try to rewrite this and start with a test first

Reply:
nm @bcardarella  it looks like I missed some parens. Will try to get a specific test for it.